File: src/dataset/Java/beer-song_aaron-contreras/gemini/GeminiFlash_BeerSong_v1.java
public class BeerSong {

    public String sing(int beers, int amountTaken) {
        int finalVerse = Math.max(0, beers - amountTaken);
        StringBuilder builder = new StringBuilder();

        for (; beers > finalVerse; beers--) {
            if (beers == 1) {
                builder.append("1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n\n");
            } else if (beers == 2) {
                builder.append("2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n\n");
            }
            else {
                builder.append(beers).append(" bottles of beer on the wall, ").append(beers).append(" bottles of beer.\nTake one down and pass it around, ").append(beers - 1).append(" bottles of beer on the wall.\n\n");
            }
        }

        if (finalVerse == 0 && amountTaken >= beers && beers != 0) {
             builder.append("No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n\n");
        }

        return builder.toString();
    }

    public String singSong() {
        return sing(99, 100);
    }
}

File: src/dataset/typescript/beer-song_Exercism-typescript-FilipeCerejo/gemini/GeminiFlash_beer-song_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.verse = verse;
exports.sing = sing;
const verseTemplate = `#begin# bottle#plural# of beer on the wall, #begin# bottle#plural# of beer.\nTake #oneit# down and pass it around, #end# bottle#endplural# of beer on the wall.\n`;
const lastVerse = `No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n`;
const replacements = {
    begin: /#begin#/g,
    end: /#end#/g,
    plural: /#plural#/g,
    endplural: /#endplural#/g,
    oneit: /#oneit#/g,
};
function verse(index) {
    if (index) {
        let result = verseTemplate;
        result = result.replace(replacements.begin, index.toString());
        result = result.replace(replacements.end, index - 1 > 0 ? (index - 1).toString() : 'no more');
        result = result.replace(replacements.plural, index === 1 ? '' : 's');
        result = result.replace(replacements.endplural, index === 2 ? '' : 's');
        result = result.replace(replacements.oneit, index === 1 ? 'it' : 'one');
        return result;
    }
    return lastVerse;
}
function sing(initialBottlesCount = 99, takeDownCount = 0) {
    const begin = initialBottlesCount;
    const end = takeDownCount;
    const result = [];
    for (let i = begin; i >= end; i--) {
        result.push(verse(i));
    }
    return result.join('\n');
}
